src/collision/AABB.js

```diff
@@ -60,7 +60,9 @@ AABB.prototype.setFromPoints = function(points, position, angle, skinSize){
     // Compute cosines and sines just once
     var cosAngle = Math.cos(angle),
         sinAngle = Math.sin(angle);
-    for(var i = 1; i<points.length; i++){
+    var len = points.length;
+    
+    for(var i = 1; i < len; i++){
```

Comment:
I'm actually not a fan of this micro-optimization, because it actually doesn't make [much of a difference](https://jsperf.com/array-length-vs-cached) to perf, but reduces readability.

I think spending time in other places that are more optimizable, like `Array.prototype.splice`. A common replacement is the [Fast & Low GC](https://jsperf.com/fast-array-splice/38) one in that jsperf, nearly double the speed boost without creating GC trash.

Looks like there even may be a fast splice function here, but it isn't used from what I can see in many of these splice locations.